### PR TITLE
Updated the model to Claude v3

### PR DESCRIPTION
**Pull request by barnesrobert**
* Updated the model to Claude v3
* Added a CloudFormation parameter for the Bedrock model ID
* Updated Lambda function runtime to 3.12

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by barnesrobert — COMMENTED** [review]
Reviewed.

**Review by rstrahan — CHANGES_REQUESTED** [review]
Hi Rob, Thanks! I left a few comments inline. 
Can you also add a few notes to the PR re:
- what models have you have tested for compatibility.. does it work with the entire claude-3 family (it should) but does it also have backward compatibility with the whole claude-2 family? (I'm not sure if all claude-2 models support the new messages api)
- Any other testing you've done to validate it.
Cheers
Bob

File: generative-ai-solutions/recordings-summary-generator/recordings-summary-generation.yaml [hunk 1]
```diff
@@ -38,6 +41,11 @@ Parameters:
     Description: These are the instructions given to the Bedrock model to generate the summary.
     Default: Provide a summary and next steps if there are any.
 
+  BedrockModelId:
```
[the remark this answers is not in the document]
Comment (barnesrobert) [reply generative-ai-solutions/recordings-summary-generator/recordings-summary-generation.yaml:44]: Good point. I was trying to make this future-friendly by leaving the value freeform, but constraining the value is a better present-friendly approach.

**Review by barnesrobert — COMMENTED** [review]
Updated the template based on reviewer recommendations.

**Review by rstrahan — APPROVED** [review]
LGTM

**Review by barnesrobert — COMMENTED** [review]
Changes reviewed.

**Comment by amcginlay** [comment]
@barnesrobert FYI I get an AccessDeniedException.
I managed to fix this by updating the `BedrockPermissions` in the `InvokeBedrockModelFunctionRole` resource as follows:

```
        - PolicyName: BedrockPermissions
          PolicyDocument:
            Version: 2012-10-17
            Statement:
              - Effect: Allow
                Action:
                  - bedrock:InvokeModel
                Resource:
                  - !Sub arn:aws:bedrock:${AWS::Region}::foundation-model/anthropic.claude-v2
                  - !Sub arn:aws:bedrock:${AWS::Region}::foundation-model/anthropic.claude-3-sonnet-20240229-v1:0
                  - !Sub arn:aws:bedrock:${AWS::Region}::foundation-model/anthropic.claude-3-haiku-20240307-v1:0
```

**Comment by barnesrobert** [comment]
Ah, thanks, @amcginlay. I created a [pull request](https://github.com/aws-samples/amazon-bedrock-samples/pull/155) to update the IAM role policy. The least-privilege IAM policy will be constrained to the selected model ID.

**Pull request merged**